w11h03/test/pgdp/networking/NetworkTest.java

```diff
@@ -0,0 +1,317 @@
+                    associated with the TUM kennung. Please open
+                    it and provide the config file with the pass-
+                    word you have received.
+                    """);
+            return;
+        }
+
+        executeTests = true;
+        System.out.println("RUNNING TESTS...");
+    }
+
+    @BeforeEach
+    public void configError() {
+        if (!executeTests)
+            fail("Please follow the instructions to set up your test files.");
+    }
+
+    //
+    // BEHAVIOUR TEST
+    //
+
+    //
+    // FLAGS
+    //
+    // [A] - Test that should <A>lawys pass and not be affected by code.
+    // [P] - Test in <P>ublic environment.
+    // [T] - <T>emplate can be broken, ensure this.
+    //
+
+    //
+    // <region structural tests>
+    //
+
+    @Test
+    @Order(1)
+    @DisplayName("[T] should be able to login with credentials")
+    public void loginTest() throws Exception {
+        assertTrue(dataHandler.login(username, password));
+
+        id = (int) Lib.getField(dataHandler, "id").get(dataHandler);
+        System.out.println("ID: " + id);
+
+        assertNotNull(id);
+        assertTrue(id > 0);
+
+        assertEquals(username, (String) Lib.getField(dataHandler, "username").get(dataHandler));
+        assertEquals(password, (String) Lib.getField(dataHandler, "password").get(dataHandler));
+    }
+
+    @Test
+    @Order(2)
+    @DisplayName("should request token")
+    public void requestTokenTest() {
+        var token = dataHandler.requestToken();
+        assertNotNull(token);
+        System.out.println("TOKEN: " + token);
+    }
+
+    @Test
+    @Order(3)
+    @DisplayName("should set visibility to public")
+    public void setVisibilityToPublic() throws Exception {
+        assertTrue(Lib.makePublic(dataHandler, true));
+    }
+
+    //
+    // </end-region>
+    // <region behaviour tests>
+    //
+
+    @Test
+    @Order(6)
+    @DisplayName("[P] should contain youself in contacts")
+    public void getContactsTest() throws Exception {
+        var users = dataHandler.getContacts();
+
+        assertNotNull(users);
+
+        // Last time I checked there were 228
+
+        assertTrue(users.size() > 100);
+
+        // Should contain self
+
+        assertTrue(users.containsKey(id));
+        assertEquals(username, users.get(id).name());
+
+        // Should contain User 1
+
+        assertTrue(users.containsKey(2032346041));
+        assertEquals("faid", users.get(2032346041).name());
+
+        // Should contain User 2
+
+        assertTrue(users.containsKey(1259660950));
+        assertEquals("nilsreichardt", users.get(1259660950).name());
+
+        // Should contain User 3
+
+        assertTrue(users.containsKey(361563966));
+        assertEquals("The One and Only", users.get(361563966).name());
+    }
+
+    @Test
+    @Order(7)
+    @DisplayName("[P] [T] should get messages from general channel")
+    public void getMessagesGeneralChatTest() throws Exception {
```

Comment:
Yes, I think the problem that the messages in "general" (id=1) has already been updated. We would need to use a channel where nobody writes messages so that the chat does not break.

Reply:
Yeah that was our conclusion too

Reply:
The test is @Disabled in latest revisions.